src/agent/src/config.rs

```diff
@@ -40,20 +40,20 @@ impl agentConfig {
         let cmdline = fs::read_to_string(file)?;
         let params: Vec<&str> = cmdline.split_ascii_whitespace().collect();
         for param in params.iter() {
-            if param.starts_with(DEBUG_CONSOLE_FLAG) {
+            if param.eq(&DEBUG_CONSOLE_FLAG) {
```

Comment:
Please can you add an extra test for each of these options to assert that, for example, `agent.devmode` set the `agentConfig` variable, but `agent.devmodefoo` doesn't.

Also, I wonder if we should make it clearer which options are simply boolean flags and which take a value. For those that take a value we could change them to be an `_OPTION` [1]:


```rust
const LOG_LEVEL_OPTION: &'static str = "agent.log";
const HOTPLUG_TIMEOUT_OPTION: &'static str = "agent.hotplug_timeout";
```

---
[1] - that would also allow us to update the existing typo nobody had noticed (`HOTPLUG_TIMOUT_FLAG` should be `HOTPLUG_TIMEOUT_FLAG` :smile:).

Reply:
Hi @jodh-intel 
Updated!